### PR TITLE
bigquery: add ML model samples

### PR DESCRIPTION
**Pull request by shollyman**

### CODE REVIEW

File: bigquery/snippets/snippet.go [hunk 1]
```diff
@@ -1325,6 +1370,18 @@ func printTableInfo(client *bigquery.Client, w io.Writer, datasetID, tableID str
...
 }
 
+func printModelInfo(client *bigquery.Client, w io.Writer, datasetID, modelID string) error {
+	ctx := context.Background()
+	// [START bigquery_get_model]
+	meta, err := client.Dataset(datasetID).Model(modelID).Metadata(ctx)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(w, "Got model '%s' with friendly name '%s'\n", modelID, meta.Name)
```
Comment (tbpg) [inline bigquery/snippets/snippet.go]: Can use `%q` to automatically quote strings.

File: bigquery/snippets/snippet.go [hunk 2]
```diff
@@ -817,6 +825,24 @@ func updateTableChangeCMEK(client *bigquery.Client, datasetID, tableID string) e
 	return nil
 }
 
+func updateModelDescription(client *bigquery.Client, datasetID, modelID string) error {
+	ctx := context.Background()
+	// [START bigquery_update_model_description]
+	model := client.Dataset(datasetID).Model(modelID)
+	oldMeta, err := model.Metadata(ctx)
+	if err != nil {
+		return err
```
Comment (tbpg) [inline bigquery/snippets/snippet.go]: Might want to add additional info to the error. https://github.com/GoogleCloudPlatform/golang-samples/blob/master/CONTRIBUTING.md#return-errors

File: bigquery/snippets/snippet_test.go [hunk 3]
```diff
@@ -502,3 +502,58 @@ func TestPartitioningAndClustering(t *testing.T) {
...
+                OPTIONS (
+                        model_type='linear_reg',
+                        max_iteration=1,
+                        learn_rate=0.4,
+                        learn_rate_strategy='constant'
+                ) AS (
+                        SELECT 'a' AS f1, 2.0 AS label
+                        UNION ALL
+                        SELECT 'b' AS f1, 3.8 AS label
+				)`, modelRef)
```
Comment (tbpg) [inline bigquery/snippets/snippet_test.go]: Weird indent here?

File: bigquery/snippets/snippet_test.go [hunk 4]
```diff
@@ -502,3 +502,58 @@ func TestPartitioningAndClustering(t *testing.T) {
...
+	_, err = job.Wait(ctx)
+	if err != nil {
+		t.Fatalf("waiting for job completion failed: %v", err)
+	}
+
+	if err := printModelInfo(client, ioutil.Discard, datasetID, modelID); err != nil {
+		t.Errorf("printModelInfo(%q %q): %v", datasetID, modelID, err)
+	}
+	if err := listModels(client, ioutil.Discard, datasetID); err != nil {
+		t.Errorf("printModelInfo(%q): %v", datasetID, err)
```
Comment (tbpg) [inline bigquery/snippets/snippet_test.go]: `listModels`

Comment (shollyman) [reply bigquery/snippets/snippet_test.go]: Done.

File: bigquery/snippets/snippet.go (hunk 2 above)
Comment (shollyman) [reply bigquery/snippets/snippet.go]: Done

File: bigquery/snippets/snippet.go (hunk 1 above)
Comment (shollyman) [reply bigquery/snippets/snippet.go]: Done

File: bigquery/snippets/snippet_test.go (hunk 3 above)
Comment (shollyman) [reply bigquery/snippets/snippet_test.go]: Yeah, something expanded oddly.  Reduced again.

File: bigquery/snippets/snippet.go [hunk 5]
```diff
@@ -831,7 +831,7 @@ func updateModelDescription(client *bigquery.Client, datasetID, modelID string)
 	model := client.Dataset(datasetID).Model(modelID)
 	oldMeta, err := model.Metadata(ctx)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to get model metadata: %v", err)
```
Comment (tbpg) [inline bigquery/snippets/snippet.go]: Usually just `return fmt.Errorf("Metadata: %v", err)`.

Same with other errors returned in this PR.

**Pull request merged**